modal/_utils/async_utils.py

```diff
@@ -544,3 +545,116 @@ async def producer(producer_id: int, iterable: Union[AsyncIterable[T], Iterable[
 
 async def callable_to_agen(awaitable: Callable[[], Awaitable[T]]) -> AsyncGenerator[T, None]:
     yield await awaitable()
+
+
+async def async_map(
+    input: Union[AsyncIterable[T], Iterable[T]],
+    async_mapper_func: Callable[[T], Awaitable[V]],
```

Comment:
If we are also allowing this to be a non-async function (as seems to be the case below), we should probably update the signature here to have `Union[Awaitable[V], V]` in the return value.

Reply:
Not sure if that fixes some typing issues too

Reply:
Never mind, I actually think we should *not* allow sync methods to be passed in here, to make the caller responsible for making the conversion to async with the foot gun that is w.r.t. to blocking the event loop